docs/index.md

```diff
@@ -0,0 +1,260 @@
+
+As the user taps or pans the map, the app either displays details for a tapped place or a `SnackBar` is displayed and a  new search is initiated for the panned location. Navigation changes are monitored by attaching a `NavigationChangedListener` to the `MapView`.  With each event received, we check the `MapView.isNavigating()` method and in the case where a user uses a [fling gesture](https://developer.android.com/training/gestures/detector.html) there is a slight pause before the fling that results in the method returning false as the API does not currently account for the delay.  To work around this, we add logic to the message queue and execute after 50 milliseconds which ensures fling gestures are captured.
+
+```java
+// This is a workaround for detecting when a fling motion has completed on the map view. The
+// NavigationChangedListener listens for navigation changes, not whether navigation has completed. We wait
+// a small interval before checking if map is view still navigating.
+
+mNavigationChangedListener = new NavigationChangedListener() {
+
+  @Override public void navigationChanged(final NavigationChangedEvent navigationChangedEvent) {
+   if (!mMapView.isNavigating()){
+     Handler handler = new Handler();
+     handler.postDelayed(new Runnable() {
+
+       @Override public void run() {
+
+         if (!mMapView.isNavigating()) {
+           onMapViewChange();
+         }
+       }
+     }, 50);
+   }
+  }
+};
+mMapView.addNavigationChangedListener(mNavigationChangedListener);
+```
+
+Each time a search is initiated from the map view, the found places are constrained to the current visible area of the map using the GeometryEngine's within method.
+
+```java
+Envelope visibleMapArea = mapView.getVisibleArea().getExtent();
+
+// Only show places within the visible map area
+Point placePoint = new Point(location.getX(),location.getY(), SpatialReferences.getWgs84());
+if (GeometryEngine.within(placePoint,visibleMapArea)){
+  places.add(place);
+}else{
+  Log.i("GeometryEngine", "***Excluding " + place.getName() + " because it's outside the visible area of map.***");
+}
+```
+
+## Routing
+
+A walking route is generated for a place by tapping on the routing arrow in the upper right hand corner of the toolbar.
+
+|Nearby Centered View|Route View|
+|----|----|
+|![Nearby Centered](./images/nearby_centered_place.png)|![Route View](./images/nearby_route_view.png)|
+
+Getting navigation directions in the nearby-app is just as easy in the [Runtime SDK](https://developers.arcgis.com/features/directions/) as it is on [ArcGIS Online](http://doc.arcgis.com/en/arcgis-online/use-maps/get-directions.htm).  A good example of how to set this up is found in the [maps-app](https://github.com/Esri/maps-app-android/blob/master/README.md) so we won't discuss it here.  In contrast to the maps-app, the nearby-app demonstrates how to set up a request for a walking route using travel modes and restrictions in the routing request.
```

Comment:
Change consistent with two other considerations above:
* "the nearby-app" --> "Nearby Places"

Again, I don't think we should hyphenate the names of these apps and "app" is part of the application's official title, so this one applies as well:
> ...is found in the [maps-app](https://github.com/Esri/maps-app-android/blob/master/README.md)

Should instead be:
> ...is found in the [Maps App](https://github.com/Esri/maps-app-android/blob/master/README.md)